FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -28,26 +27,37 @@ class Flagsmith constructor(
     private val baseUrl: String = "https://edge.api.flagsmith.com/api/v1",
     private val context: Context? = null,
     private val enableAnalytics: Boolean = DEFAULT_ENABLE_ANALYTICS,
-    private val analyticsFlushPeriod: Int = DEFAULT_ANALYTICS_FLUSH_PERIOD_SECONDS
+    private val enableCache: Boolean = DEFAULT_ENABLE_CACHE,
```

Comment:
Hi @matthewelwell when I originally did this with Fuse HTTP the user could pass-in a 'Cache' implementation and do what they want. This would have covered all bases, and I just provided a static `deafaultCache()` method on the SDK to use the deafult.

On OKHttp the Cache class is what it is, you can't override it or replace it.

`public final class Cache implements Closeable, Flushable {`

Mobile apps are sandbox encrypted at the hardware-level now, there's no real need to double-encrypt it yourself.

The Cache implements two constructors:

```
  public Cache(File directory, long maxSize) {
    this(directory, maxSize, FileSystem.SYSTEM);
  }

  Cache(File directory, long maxSize, FileSystem fileSystem) {
    this.cache = DiskLruCache.create(fileSystem, directory, VERSION, ENTRY_COUNT, maxSize);
  }
```

The FileSystem one seems to be to allow for unit testing of OKHttp itself and not particularly useful to create your own Filesystem.

Also worth bearing in mind that the cache config isn't even in the Cache class:

```
                .callTimeout(REQUEST_TIMEOUT_SECONDS, java.util.concurrent.TimeUnit.SECONDS)
                .readTimeout(READ_WRITE_TIMEOUT_SECONDS, java.util.concurrent.TimeUnit.SECONDS)
                .writeTimeout(READ_WRITE_TIMEOUT_SECONDS, java.util.concurrent.TimeUnit.SECONDS)
                .cache(if (context != null && enableCache) okhttp3.Cache(context.cacheDir, cacheSize) else null)
```

So there's not a lot of point injecting one as you can't inject one with custom timeouts or whatever.

Reply:
I have created a FlagsmithCacheConfig class to wrap up all of the config and defaults @matthewelwell